test/linalg/dense.jl

```diff
@@ -582,6 +582,25 @@ end
     end
 end
 
+@testset "issue #23366 (Int Matrix to Int power)" begin
+    @testset "Tests for $elty" for elty in (Int128, Int16, Int32, Int64, Int8,
+                                            UInt128, UInt16, UInt32, UInt64, UInt8,
+                                            BigInt)
+        @test_throws DomainError elty[1 1;1 0]^-2
+        @test elty[1 1;1 0]^2 == elty[2 1;1 1]
```

Comment:
Yes, you're right, I'll adjust that. (I'm not a big fan of `@inferred` in unit tests, since it results in confusing test failure messages, see https://discourse.julialang.org/t/more-informative-inferred-especially-for-unit-testing/5481 . But I guess there's no better solution at the moment.)